### PR TITLE
Honour the provided scripts if they have been uploaded

### PR DESCRIPTION
**Pull request by sihil**
We've been throwing away any uploaded upstart/systemd config files for ages. We should really honour them if people _really_ want to supply a config file.

This might cause confusion when people switch over to this if they have a config file that exists but hasn't been used - this change will start to pick it up. I could make this an option so that we retain the existing behaviour by default if that would be considered 'better'.

I'm principally changing this for my own selfish reasons as I need to provide a custom file for riff-raff for self deployment.

### CODE REVIEW

**Comment by sihil** [comment]
@cb372

**Comment by cb372** [comment]
Looks reasonable.

> This might cause confusion when people switch over to this

I wouldn't worry about backwards compatibility too much. As far as I know, nobody is using this feature yet apart from you.

Would be interesting to hear more about why you needed to customise the startup script. If the default script provided by the feature is too simplistic and everyone ends up using a custom script, maybe there's no need to provide a default at all.

**Comment by sihil** [comment]
The reason for customisation is the way that Riff-Raff self-deploys. In the new AWS world we will upload the new version to S3 and then turn a switch on Riff-Raff's management page to ON. This causes Riff-Raff to exit with code 219 as soon as it is no longer deploying. The config file is then written to re-download and restart the app in this circumstance.

This is very much a special snowflake deployment approach and I wouldn't expect any other nodes to use this approach so I think this is very much the exception rather than the rule.

To put it another way, I've used this feature dozens of times elsewhere and haven't needed to create a custom config file. However various projects already have a custom file which will by default be included in the deployment artifact and uploaded to S3. This is currently ignored if they use this feature, but would be picked up.

**Comment by sihil** [comment]
Having thought more about it I could just use an entirely custom boot process for Riff-Raff instead of using this feature. This probably isn't a bad idea as I want to download from S3 in systemd and not during the `install.sh` phase.
